operate/services/manage.py

```diff
@@ -1362,6 +1389,7 @@ def deploy_service_locally(self, hash: str, force: bool = True) -> Deployment:
         :param force: Remove previous deployment and start a new one.
         :return: Deployment instance
         """
+        self._set_env_variables(hash=hash)
```

Comment:
@0xArdi Although this approach is far from ideal, I think this is a cleaner, less error-prone way than what we had on [`Deployment._build_host`](https://github.com/valory-xyz/olas-operate-app/blob/main/operate/services/service.py#L581-L590) and [`Deployment._build_docker`](https://github.com/valory-xyz/olas-operate-app/blob/main/operate/services/service.py#L482-L494). 

I wonder why this approach was implemented this way (would the override as environment variable cause any issue?) Let me know any comments.